Issue while install py-libsonata

We encountered this issue while deploying on Jureca system @ JSC. Adding details here for future reference:

```
○ → spack spec -I py-libsonata@0.0.2%gcc
Input spec
--------------------------------
 -   py-libsonata@0.0.2%gcc

Concretized
--------------------------------
 -   py-libsonata@0.0.2%gcc@8.3.0 arch=linux-centos7-x86_64 
[+]      ^cmake@3.13.0%gcc@8.3.0~doc+ncurses+openssl+ownlibs~qt arch=linux-centos7-x86_64 
[+]      ^hdf5@1.10.4%gcc@8.3.0~cxx~debug~fortran+hl~mpi+pic+shared~szip~threadsafe arch=linux-centos7-x86_64 
 -       ^py-numpy@1.15.2%gcc@8.3.0+blas+lapack arch=linux-centos7-x86_64 
[+]          ^intel-mkl@2019.1.144%gcc@8.3.0~ilp64+shared threads=none arch=linux-centos7-x86_64 
[+]          ^py-setuptools@40.4.3%gcc@8.3.0 arch=linux-centos7-x86_64 
[+]              ^python@2.7.15%gcc@8.3.0+dbm~optimizations patches=123082ab3483ded78e86d7c809e98a804b3465b4683c96bd79a2fd799f572244 +pic+pythoncmd+shared~tk~ucs4 arch=linux-centos7-x86_64 

# and spack install gives

==> Installing py-libsonata
==> Searching for binary cache of py-libsonata
==> Warning: No Spack mirrors are currently configured
==> No binary for py-libsonata found: installing from source
==> Fetching https://pypi.io/packages/source/l/libsonata/libsonata-0.0.2.tar.gz
######################################################################## 100.0%
==> Staging archive: /p/project/cvsk25/blancoalonso1/HBP/jureca-cluster/27-05-2019/sources/spack/var/spack/stage/py-libsonata-0.0.2-ychfl66bo3x73snurf3zj6qphsxxliti/libsonata-0.0.2.tar.gz
==> Created stage in /p/project/cvsk25/blancoalonso1/HBP/jureca-cluster/27-05-2019/sources/spack/var/spack/stage/py-libsonata-0.0.2-ychfl66bo3x73snurf3zj6qphsxxliti
==> No patches needed for py-libsonata
==> Building py-libsonata [PythonPackage]
==> Executing phase: 'build'
==> '/usr/local/software/jureca/Stages/2019a/software/Python/2.7.16-GCCcore-8.3.0/bin/python2.7' '-s' 'setup.py' '--no-user-cfg' 'build'
.....
[100%] Built target sonata_python
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/egg
copying build/lib.linux-x86_64-2.7/libsonata.so -> build/bdist.linux-x86_64/egg
creating stub loader for libsonata.so
byte-compiling build/bdist.linux-x86_64/egg/libsonata.py to libsonata.pyc
creating build/bdist.linux-x86_64/egg/EGG-INFO
copying libsonata.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying libsonata.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying libsonata.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying libsonata.egg-info/not-zip-safe -> build/bdist.linux-x86_64/egg/EGG-INFO
copying libsonata.egg-info/requires.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying libsonata.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
writing build/bdist.linux-x86_64/egg/EGG-INFO/native_libs.txt
creating dist
creating 'dist/libsonata-0.0.2-py2.7-linux-x86_64.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Creating /p/project/cvsk25/blancoalonso1/HBP/jureca-cluster/27-05-2019/install/linux-centos7-x86_64/gcc-8.3.0/py-libsonata-0.0.2-ychfl6/lib/python2.7/site-packages/site.py
Processing libsonata-0.0.2-py2.7-linux-x86_64.egg
creating /p/project/cvsk25/blancoalonso1/HBP/jureca-cluster/27-05-2019/install/linux-centos7-x86_64/gcc-8.3.0/py-libsonata-0.0.2-ychfl6/lib/python2.7/site-packages/libsonata-0.0.2-py2.7-linux-x86_64.egg
Extracting libsonata-0.0.2-py2.7-linux-x86_64.egg to /p/project/cvsk25/blancoalonso1/HBP/jureca-cluster/27-05-2019/install/linux-centos7-x86_64/gcc-8.3.0/py-libsonata-0.0.2-ychfl6/lib/python2.7/site-packages
Adding libsonata 0.0.2 to easy-install.pth file

Installed /p/project/cvsk25/blancoalonso1/HBP/jureca-cluster/27-05-2019/install/linux-centos7-x86_64/gcc-8.3.0/py-libsonata-0.0.2-ychfl6/lib/python2.7/site-packages/libsonata-0.0.2-py2.7-linux-x86_64.egg
Processing dependencies for libsonata==0.0.2
Searching for numpy>=1.12.0
Reading https://pypi.python.org/simple/numpy/
Best match: numpy 1.16.3
Downloading https://files.pythonhosted.org/packages/93/48/956b9dcdddfcedb1705839280e02cbfeb2861ed5d7f59241210530867d5b/numpy-1.16.3.zip#sha256=78a6f89da87eeb48014ec652a65c4ffde370c036d780a995edaeb121d3625621
Processing numpy-1.16.3.zip
Writing /tmp/easy_install-UpVAxi/numpy-1.16.3/setup.cfg
Running numpy-1.16.3/setup.py -q bdist_egg --dist-dir /tmp/easy_install-UpVAxi/numpy-1.16.3/egg-dist-tmp-onIeps
Running from numpy source directory.
.....
Warning: distutils distribution has been initialized, it may be too late to add an extension lapack_liteWarning: distutils distribution has been initialized, it may be too late to add an extension _umath_linalgWarning: distutils distribution has been initialized, it may be too late to add a subpackage linalgWarning: distutils distribution has been initialized, it may be too late to add a subpackage maWarning: distutils distribution has been initialized, it may be too late to add a subpackage matrixlibWarning: distutils distribution has been initialized, it may be too late to add a subpackage polynomialWarning: distutils distribution has been initialized, it may be too late to add an extension mtrandWarning: distutils distribution has been initialized, it may be too late to add a subpackage randomWarning: distutils distribution has been initialized, it may be too late to add a subpackage _privateWarning: distutils distribution has been initialized, it may be too late to add a subpackage testingWarning: distutils distribution has been initialized, it may be too late to add a subpackage numpy_configtest.c:1:5: warning: conflicting types for built-in function ‘exp’ [-Wbuiltin-declaration-mismatch]
 int exp (void);
     ^~~
_configtest.c:6: error: undefined reference to 'exp'
collect2: error: ld returned 1 exit status
_configtest.c:1:5: warning: conflicting types for built-in function ‘exp’ [-Wbuiltin-declaration-mismatch]
 int exp (void);
     ^~~
_configtest.c:1:10: fatal error: sys/endian.h: No such file or directory
 #include <sys/endian.h>
          ^~~~~~~~~~~~~~
compilation terminated.
_configtest.c: In function ‘main’:
_configtest.c:5:16: warning: variable ‘test_array’ set but not used [-Wunused-but-set-variable]
     static int test_array [1 - 2 * !(((long) (sizeof (npy_check_sizeof_type))) >= 0)];
                ^~~~~~~~~~
_configtest.c: In function ‘main’:
_configtest.c:5:16: warning: variable ‘test_array’ set but not used [-Wunused-but-set-variable]
     static int test_array [1 - 2 * !(((long) (sizeof (npy_check_sizeof_type))) == 4)];
                ^~~~~~~~~~
```

we need to look into more but `setup.py` internally trying to install another `numpy` is causing this pain (as @pramodk  said).

cc: @pramodk

--- a/var/spack/repos/builtin/packages/libsonata/package.py
+++ b/var/spack/repos/builtin/packages/libsonata/package.py
@@ -19,8 +19,7 @@ class Libsonata(CMakePackage):
     git = "https://github.com/BlueBrain/libsonata.git"
 
     version('develop', branch='master', submodules=False)
-    version('0.0.3', commit='fd77f22d94737593f8ac86aa762d25a8b8622100', submodules=False)
-    version('0.0.2', commit='3ebd44e802b5b46008b255ec0c1e2f0eabdf6ac6', submodules=False)
+    version('0.0.3', tag='v0.0.3', submodules=False)
 
     variant('mpi', default=False, description="Enable MPI backend")
 

--- a/var/spack/repos/builtin/packages/py-bglibpy/package.py
+++ b/var/spack/repos/builtin/packages/py-bglibpy/package.py
@@ -9,12 +9,12 @@ from spack import *
 class PyBglibpy(PythonPackage):
     """Pythonic Blue Brain simulator access"""
     homepage = "https://bbpcode.epfl.ch/code/#/admin/projects/sim/BGLibPy"
+    url = "ssh://bbpcode.epfl.ch/sim/BGLibPy"
     git = "ssh://bbpcode.epfl.ch/sim/BGLibPy"
 
     version('develop', branch='master')
-    version(
-        '4.0.27',
-        commit='42d9c1f891ef1ec9af6d72c49ff3b7726a009951')
+    version('4.0.27', commit='42d9c1f891ef1ec9af6d72c49ff3b7726a009951')
+    version('4.0.17', commit='e90513f52a7d9ca3c16877ae7a2bcb8df31c8545')
 
     depends_on('py-setuptools', type=('build', 'run'))
 

--- a/var/spack/repos/builtin/packages/py-libsonata/package.py
+++ b/var/spack/repos/builtin/packages/py-libsonata/package.py
@@ -12,8 +12,7 @@ class PyLibsonata(PythonPackage):
     homepage = "https://github.com/BlueBrain/libsonata"
     git = "https://github.com/BlueBrain/libsonata.git"
 
-    version('0.0.3', commit='fd77f22d94737593f8ac86aa762d25a8b8622100', submodules=True)
-    version('0.0.2', commit='3ebd44e802b5b46008b255ec0c1e2f0eabdf6ac6', submodules=True)
+    version('0.0.3', tag='v0.0.3', submodules=True)
 
     depends_on('cmake@3.3:', type='build')
     depends_on('hdf5~mpi', type='build')

--- a/var/spack/repos/builtin/packages/py-libsonata/package.py
+++ b/var/spack/repos/builtin/packages/py-libsonata/package.py
@@ -18,4 +18,4 @@ class PyLibsonata(PythonPackage):
     depends_on('cmake@3.3:', type='build')
     depends_on('hdf5~mpi', type='build')
 
-    depends_on('py-numpy@1.12:', type='run')
+    depends_on('py-numpy@1.12:', type=('build', 'run'))

